FlipViceFeed/app/src/main/java/com/joelimyx/flipvicefeed/Notifications/AlarmReceiver.java

```diff
@@ -0,0 +1,98 @@
+package com.joelimyx.flipvicefeed.notifications;
+
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.PersistableBundle;
+import android.support.v7.app.NotificationCompat;
+import android.util.Log;
+import android.widget.RemoteViews;
+
+import com.joelimyx.flipvicefeed.R;
+import com.joelimyx.flipvicefeed.classes.TopicObject;
+import com.joelimyx.flipvicefeed.database.AlarmSQLHelper;
+import com.joelimyx.flipvicefeed.main.main.MainActivity;
+
+import java.util.List;
+
+import static android.content.ContentValues.TAG;
+
+/**
+ * Created by KorbBookProReturns on 11/30/16.
+ */
+
+public class AlarmReceiver extends BroadcastReceiver {
+    public static final int NOTIFICATION_ID = 0;
+    public static final String PACKAGE_NAME = "com.joelimyx.flipvicefeed";
+    public static final String NEW_STORIES_DEFAULT = "Check out the latest stories!";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        //-------------------------------
+        //  When an alarm is received, if the api is 21 or above, it sends it to a job scheduler
+        //  so it pulls the latest article's image and uses that as the background of the notification.
+        //  But it'll only show a notification if there's internet, and it'll try again every 15 minutes.
+        //
+        //  ALSO, if there are no topics favorited, then it just runs a default notification.
+        //-------------------------------
+
+        List<TopicObject> favoriteTopics = AlarmSQLHelper.getInstance(context).getFavoriteTopics();
+
+        if (Build.VERSION.SDK_INT >= 21 && !favoriteTopics.isEmpty()) {
```

Comment:
Good job handling the case when the device is running SDK < 21 - however, you only used JobScheduler for one small feature, and did not use it to schedule any background loading of articles for display in the main screens of the app.